src/cmd/delegation_backend/main_bpu.go

```diff
@@ -29,23 +29,38 @@ func main() {
 
 	appCfg := LoadEnv(log)
 
-	awsCfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(appCfg.Aws.Region))
-	if err != nil {
-		log.Fatalf("Error loading AWS configuration: %v", err)
-	}
-
 	app := new(App)
 	app.Log = log
 	http.HandleFunc("/", func(rw http.ResponseWriter, r *http.Request) {
 		_, _ = rw.Write([]byte("delegation backend service"))
 	})
 	http.Handle("/v1/submit", app.NewSubmitH())
-	client := s3.NewFromConfig(awsCfg)
 
-	awsctx := AwsContext{Client: client, BucketName: aws.String(GetBucketName(appCfg)), Prefix: appCfg.NetworkName, Context: ctx, Log: log}
-	app.Save = func(objs ObjectsToSave) {
-		awsctx.S3Save(objs)
+	if appCfg.Aws != nil {
```

Comment:
I was thinking that this should be before the `http.Handle` call. The app should be configured before the backend function. Does this work like this? Did you try to run the `delegation_backend` go package?

Reply:
Yes, I tried it. Integration test works as before, so it is writing to the AWS. I think it doesn't make much difference because the server is started in the last line of the `main` function with `log.Fatal(http.ListenAndServe(...))` and `http.Handle` is just setting the route. So even if the HTTP routes are registered before the storage backends are initialized, no requests will be processed until the server is actually started.

However for clearer logical flow we may put it before :+1:

Reply:
Thanks for the explanation, makes much more sense now!